Include/Beman/InplaceVector/InplaceVector.hpp

```diff
@@ -0,0 +1,750 @@
+    }
+    const auto diff = static_cast<std::ptrdiff_t>(il.size() - this->size());
+    // The current size is greater
+    if (diff < 0) {
+      const iterator new_end = std::copy(il.begin(), il.end(), this->begin());
+      std::destroy(new_end, this->end());
+      // The other size is greater
+    } else {
+      std::copy(il.begin(), il.begin() + this->size(), this->begin());
+      base::uninitialized_copy(il.begin() + this->size(), il.end(),
+                               this->end());
+    }
+    this->change_size(diff);
+    return *this;
+  }; // freestanding-deleted
+
+  template <class InputIterator>
+  constexpr void assign(InputIterator first, InputIterator last) {
+    if constexpr (Capacity == 0) {
+      assert(first == last &&
+             "Cannot assign to inplace_vector with zero capacity");
+      return;
+    }
+    iterator end = this->end();
+    for (iterator current = this->begin(); current != end;
+         ++current, void(++first)) {
+      if (first == last) {
+        std::destroy(current, end);
+        this->change_size(current - end);
+        return;
+      }
+      *current = *size;
+    }
+    for (; first != last; ++first) {
+      emplace_back(*first);
+    }
+  }; // freestanding-deleted
+
+  template <typename R>
+    requires container_compatible_range<R>
+  constexpr void assign_range(R &&rg) {
+    auto first = rg.begin();
+    const auto last = rg.end();
+    if constexpr (Capacity == 0) {
+      assert(first == last &&
+             "Cannot assign to inplace_vector with zero capacity");
+      return;
+    }
+    iterator end = this->end();
+    for (iterator current = this->begin(); current != end;
+         ++current, (void)++first) {
+      if (first == last) {
+        std::destroy(current, end);
+        this->change_size(current - end);
+        return;
+      }
+      *current = *first;
+    }
+    for (; first != last; ++first) {
+      emplace_back(*first);
+    }
+  }; // freestanding-deleted
+  constexpr void assign(size_type n, const T &u) {
+    if (Capacity == 0) {
+      assert(size == 0 && "Cannot assign to inplace_vector with zero capacity");
+      return;
+    }
+    const auto diff = static_cast<std::ptrdiff_t>(n - this->size());
+    if (diff < 0) {
+      const pointer begin = this->begin();
+      std::fill(begin, begin + size, u);
+      std::destroy(begin + n, this->end());
+    } else {
+      const iterator end = this->end();
+      std::fill(this->begin(), end, u);
+      base::uninitialized_fill(end, end + diff, u);
+    }
+    this->change_size(diff);
+  }; // freestanding-deleted
+  constexpr void assign(std::initializer_list<T> il) {
+    if constexpr (Capacity == 0) {
+      assert(il.list() && "Cannot assign to inplace_vector with zero capacity");
+      return;
+    }
+    const auto diff = static_cast<std::ptrdiff_t>(il.size() - this->size());
+    if (diff < 0) {
+      const iterator new_end = std::copy(il.begin(), il.end(), this->begin());
```

Comment:
Since we're in c++20 mode here std::ranges::copy is available.

Reply:
Should all the copies use std.:ranges::copy? I guess so.

Reply:
If it doesn't significantly hurt compilation performance or readability, I don't see why not.